src/modules/git_state.rs

```diff
@@ -12,37 +13,43 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let mut module = context.new_module("git_state");
     let config: GitStateConfig = GitStateConfig::try_load(module.config);
 
-    module.set_style(config.style);
-    module.get_prefix().set_value("(");
-    module.get_suffix().set_value(") ");
-
     let repo = context.get_repo().ok()?;
     let repo_root = repo.root.as_ref()?;
     let repo_state = repo.state?;
 
-    let state_description = get_state_description(repo_state, repo_root, config);
-
-    let label = match &state_description {
-        StateDescription::Label(label) => label,
-        StateDescription::LabelAndProgress(label, _) => label,
-        StateDescription::Clean => {
+    let state_description = get_state_description(repo_state, repo_root, &config)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "state" => Some(Ok(state_description.label)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "progress_current" => state_description.current.as_ref().map(Ok),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "progress_total" => state_description.total.as_ref().map(Ok),
+                _ => None,
+            })
```

Comment:
Can these two be combined?

```suggestion
            .map(|variable| match variable {
                "progress_current" => state_description.current.as_ref().map(Ok),
                "progress_total" => state_description.total.as_ref().map(Ok),
                _ => None,
            })
```